eng/common/TestResources/New-TestResources.ps1

```diff
@@ -743,13 +752,15 @@ try {
     if ($ProvisionerApplicationOid) {
         $templateParameters["provisionerApplicationOid"] = "$ProvisionerApplicationOid"
     }
-
     if ($TenantId) {
         $templateParameters.Add('tenantId', $TenantId)
     }
     if ($TestApplicationSecret -and $ServicePrincipalAuth) {
         $templateParameters.Add('testApplicationSecret', $TestApplicationSecret)
     }
+    if ($CI -and $Environment -eq 'AzureCloud') {
```

Comment:
Should we do this based on cloud or tenant?

Reply:
Cross tenant vnet linking will work, but cross cloud will not, is my thinking (since the vnet is not multi cloud). So we would have to filter on public cloud only. Sovereign clouds aren't a problem for us since they are already in an isolated tenant. For public cloud it doesn't matter which tenant because the agent request will always originate from a vnet.